crates/virtio-queue/src/lib.rs

```diff
@@ -74,13 +80,25 @@ impl Display for Error {
             InvalidIndirectDescriptor => write!(f, "invalid indirect descriptor"),
             InvalidIndirectDescriptorTable => write!(f, "invalid indirect descriptor table"),
             InvalidDescriptorIndex => write!(f, "invalid descriptor index"),
+        }
+    }
+}
+
+#[cfg(any(test, feature = "test-utils"))]
+impl Display for MockError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::MockError::*;
+
+        match self {
             InvalidIndex => write!(f, "invalid ref index"),
             InvalidNextAvail => write!(f, "invalid next avail"),
```

Comment:
Can you use clearer error messages here? `ref` is very vague for example.

Reply:
Done.